doc/weakly_supervised.rst

```diff
@@ -228,8 +312,31 @@ Residual
 SDML
 ----
 
-`SDML`: An efficient sparse metric learning in high-dimensional space via
-L1-penalized log-determinant regularization
+Sparse High-Dimensional Metric Learning
+(:py:class:`SDML <metric_learn.sdml.SDML>`)
+
+`SDML` is an efficient sparse metric learning in high-dimensional space via 
+double regularization: an L1-penalization on the off-diagonal elements of the 
+Mahalanobis matrix :math:`\mathbf{M}`, and a log-determinant divergence between 
+:math:`\mathbf{M}` and :math:`\mathbf{M_0}` (set as either :math:`\mathbf{I}` 
+or :math:`\mathbf{\Omega}^{-1}`, where :math:`\mathbf{\Omega}` is the 
+covariance matrix).
+
+The formulated optimization on the semidefinite matrix :math:`\mathbf{M}` 
+is convex:
+
+.. math::
+
+    \min_{\mathbf{M}} = \text{tr}((\mathbf{M}_0 + \eta \mathbf{XLX}^{T})
+    \cdot \mathbf{M}) - \log\det \mathbf{M} + \lambda ||\mathbf{M}||_{1, off}
+
+where :math:`\mathbf{X}=[\mathbf{x}_1, \mathbf{x}_2, ..., \mathbf{x}_n]` is 
+the training data, incidence matrix :math:`\mathbf{K}_{ij} = 1` if 
+:math:`(\mathbf{x}_i, \mathbf{x}_j)` is a similar pair, otherwise -1. The 
+Laplacian matrix :math:`\mathbf{L}` is calculated from :math:`\mathbf{K}` 
```

Comment:
I think it is better to give the definition of K, even though it requires to define D. Without this, it is hard to understand

To be concise you could write something like:
The Laplacian matrix :math:`\mathbf{L}=\mathbf{D}-\mathbf{K}` is calculated from :math:`\mathbf{K}` and :math:`\mathbf{D}`, a diagonal matrix whose entries are the sums of the row elements of :math:`\mathbf{K}`.

Reply:
done